warp10/src/main/java/io/warp10/continuum/gts/GTSHelper.java

```diff
@@ -5684,12 +5684,31 @@ public static List<GeoTimeSerie> map(GeoTimeSerie gts, Object mapper, long prewi
     // Retrieve ticks if GTS is not bucketized.
     long[] ticks = isBucketized(gts) ? null : Arrays.copyOf(mapped.ticks, gts.values);
 
-    // Sort outputTicks
+    // Check if outputTicks is correctly sorted
+    // In case of a concurrent execution, sorting outputTicks here will lead to a ConcurrentModificationException
     if (null != outputTicks) {
-      if (reversed) {
-        Collections.sort(outputTicks, Collections.<Long>reverseOrder());
-      } else {
-        Collections.sort(outputTicks);
+      if (outputTicks.size() > 1) {
+        if (reversed) {
+          boolean descending = true;
+          int i = 1;
+          while (i < outputTicks.size() && descending) {
+            descending = outputTicks.get(i) <= outputTicks.get(i - 1);
```

Comment:
get does a rangecheck so it is better if you keep track of the previous element to avoid calling get twice with the same index.